slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -28,15 +28,87 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
                         "GetMap is not yet implemented.");
 }
 
-void SLAMServiceImpl::CreateMap() {
-    mapping::MapBuilder mapBuilder;
+SLAMServiceActionMode SLAMServiceImpl::GetActionMode() {
+    // TODO: Add a case for updating. Requires that an apriori
+    // map is detected and loaded. Will be implemented in this ticket:
+    // https://viam.atlassian.net/browse/DATA-114
+
+    // TODO: Change this depending on the outcome of this scope
+    // doc:
+    // https://docs.google.com/document/d/1RsT-c0QOtkMKa-rwUGY0-emUmKIRSaO3mzT1v6MtFjk/edit#heading=h.tcicyojyqi6c
+    if (map_rate_sec.count() == -1) {
+        LOG(INFO) << "Running in localization only mode";
+        return SLAMServiceActionMode::LOCALIZING;
+    }
+    LOG(INFO) << "Running in mapping mode";
+    return SLAMServiceActionMode::MAPPING;
```

Comment:
Mapping mode is the default - unless an apriori map is loaded, or `map_rate_sec` is set to the value where we define that we're running SLAM in localization only "action mode".